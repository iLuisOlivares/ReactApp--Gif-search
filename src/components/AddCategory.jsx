import React, { useState } from 'react';
import PropTypes from 'prop-types';

function AddCategory({ setCategories }) {

    const [inputValue, setInputValue] = useState("");


    const handleOnChange = (e) => {
        setInputValue(e.target.value);
    }
    const handleSubmit = (e) => {
        e.preventDefault();

        console.log('HandleSubmit:', inputValue);
        if (inputValue.trim().length > 2) {
            setCategories(cats => [e.target[0].value, ...cats]);
            setInputValue('');
        }

    }

    return (
        <form className='search' onSubmit={handleSubmit}>
            <input

                type="text"
                value={inputValue}
                onChange={handleOnChange}
            ></input>
        </form>);

}

export default AddCategory;

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}