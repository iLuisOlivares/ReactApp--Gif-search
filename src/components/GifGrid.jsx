import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import { getGifs } from '../helpers/getGif';
import { useFetchGif } from '../hooks/useFetchGif';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {

    // const [images, setimages] = useState([]);

    // useEffect(() => {
    //     getGifs(category).then(img => setimages(img));

    // }, [category]);

    const { data: images, loading } = useFetchGif(category);

    return (
        <>
            <h3 className='animate__animated animate__fadeInUp texto'>{category}</h3>

            {/* {loading && <p className='animate__animated animate__flash'>Loading</p>} */}


            <div className='card-grid '>
                {
                    images.map((img) => (
                        <GifGridItem
                            key={img.id}
                            {...img}
                        ></GifGridItem>
                    ))
                }
            </div>
        </>
    )
}


GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}