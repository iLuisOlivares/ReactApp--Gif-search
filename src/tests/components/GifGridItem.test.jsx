import { shallow } from 'enzyme';
import React from 'react';
import { GifGridItem } from '../../components/GifGridItem';


describe('Prueba de GifGridItem', () => {

    const title = 'Title por defecto';
    const url = 'Url por defecto';
    const wrapper = shallow(<GifGridItem title={title} url={url}></GifGridItem>);

    test('Debe mostrar el componente correctamente', () => {

        expect(wrapper).toMatchSnapshot();

    });

    test('Debe mostrar un parrafo con el title', () => {
        const p = wrapper.find('p');

        expect(p.text().trim()).toBe(title);
    });

    test('Debe mostrar una imagen de la url', () => {
        const img = wrapper.find('img');

        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
    });

    test('Debe mostrar la clase animate__fadeInUp del div', () => {
        const div = wrapper.find('div');

        expect(div.prop('className').includes('animate__fadeInUp')).toBe(true);
    });

});
