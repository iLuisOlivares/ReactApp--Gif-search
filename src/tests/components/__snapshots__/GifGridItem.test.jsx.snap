// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba de GifGridItem Debe mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeInUp"
>
  <p>
    Title por defecto
  </p>
  <img
    alt="Title por defecto"
    src="Url por defecto"
  />
</div>
`;
